### PR TITLE
Add language icon to runechat

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Runechat will now show what language you're speaking in where appropriate.

[Demo here](https://file.house/mtVD.mp4)

I think most of these are good enough to fit their job as "tell from a glance the language", except for chimp talk which ends up looking bad because of the outline maptext provides. I don't consider this a blocker.

## Why It's Good For The Game
Gives people less reason to need to look away from runechat.

## Changelog
:cl:
add: Runechat messages will now also display the icon.
/:cl:

### CODE REVIEW

File: code/datums/chatmessage.dm [hunk 1]
```diff
@@ -129,13 +133,24 @@
...
 	else if (extra_classes.Find("emote"))
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "emote")
-		text =  "\icon[r_icon]&nbsp;[text]"
+		LAZYADD(prefixes, "\icon[r_icon]")
+
+	// Append language icon if the language uses one
+	var/datum/language/language_instance = GLOB.language_datum_instances[language]
+	if (language_instance?.display_icon(owner))
+		var/icon/language_icon = icon(language_instance.icon, icon_state = language_instance.icon_state)
+		language_icon.Scale(CHAT_MESSAGE_ICON_SIZE, CHAT_MESSAGE_ICON_SIZE)
```
Comment (Jared-Fogle) [inline code/datums/chatmessage.dm:150]: I don't know much about BYOND performance, but something about this gives me the chills. Should I be concerned? Should this just be cached, or is there no notable performance gain to be had?

**Comment by ATH1909** [comment]
make sure this works right with stuff like codespeak

File: code/datums/chatmessage.dm [hunk 2]
```diff
@@ -129,13 +133,24 @@
...
-		text =  "\icon[r_icon]&nbsp;[text]"
+		LAZYADD(prefixes, "\icon[r_icon]")
 	else if (extra_classes.Find("emote"))
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "emote")
-		text =  "\icon[r_icon]&nbsp;[text]"
+		LAZYADD(prefixes, "\icon[r_icon]")
+
+	// Append language icon if the language uses one
+	var/datum/language/language_instance = GLOB.language_datum_instances[language]
+	if (language_instance?.display_icon(owner))
```
Comment (Jared-Fogle) [inline code/datums/chatmessage.dm:148]: @ATH1909 It does.

**Comment by kriskog** [comment]
testmerged on terry

**Comment by kriskog** [comment]
Icons dont really work that well, but its better than nothing.

**Comment by Jared-Fogle** [comment]
@kriskog Do they look differently than my preview video, or are you just saying that?

**Comment by kriskog** [comment]
Same. Theres at least three that looks like ballsacks at that scale. 🤡

**Comment by Jared-Fogle** [comment]
image

File: code/datums/chatmessage.dm (hunk 1 above)
Comment (Cyberboss) [reply code/datums/chatmessage.dm:150]: @AnturK ? @SpaceManiac ? Someone needs to answer this because I'm concerned about it as well

**Comment by LemonInTheDark** [comment]
they just weren't intended for that scale, so it looks janky and squashed. I get the idea, but I feel like this is the wrong way to go about it

**Comment by Jared-Fogle** [comment]
I don't really have the bandwidth to create 9x9 language icons for everything, and I sincerely believe this is better than nothing (as skoglol said).

File: code/datums/chatmessage.dm (hunk 1 above)
Comment (AnturK) [reply code/datums/chatmessage.dm:150]: Runechat already has performance issues at higher pop, adding blend operations to it sounds like horrible idea.

Comment (Cyberboss) [reply code/datums/chatmessage.dm:150]: Can we cache these scalings and send it as an asset?

Comment (AnturK) [reply code/datums/chatmessage.dm:150]: This or using transform at least.

**Comment by LemonInTheDark** [comment]
That's true, it's a shame, but you're right.

**Comment by Jared-Fogle** [comment]
On the flip side, maybe this'll anger a spriter enough to make good icons 😈

**Pull request merged**